tests/unittests/OnnxImporterTest.cpp

```diff
@@ -27,6 +27,144 @@
+      auto RH = tensors[i]->getHandle<>();
+      ONNX_NAMESPACE::TensorProto *tp = gp->add_initializer();
+      tp->set_name(tensorNames[i]);
+      for (int k = 0; k < tensors[i]->dims().size(); k++) {
+        tp->add_dims(tensors[i]->dims()[k]);
+      }
+      switch (RH.getElementType()) {
+      case ElemKind::FloatTy:
+        tp->set_data_type(ONNX_NAMESPACE::TensorProto::FLOAT);
+        for (int k = 0; k < tensors[i]->size(); k++) {
+          tp->add_float_data(RH.raw(k));
+        }
+        break;
+      case ElemKind::Int64ITy:
+        tp->set_data_type(ONNX_NAMESPACE::TensorProto::INT64);
+        for (int k = 0; k < tensors[i]->size(); k++) {
+          tp->add_int64_data(RH.raw(k));
+        }
+        break;
+      case ElemKind::Int32ITy:
+        tp->set_data_type(ONNX_NAMESPACE::TensorProto::INT32);
+        for (int k = 0; k < tensors[i]->size(); k++) {
+          tp->add_int32_data(RH.raw(k));
+        }
+        break;
+      default:
+        std::cout << "Unsupported datatype";
+        return false;
+      }
+    }
+  }
+  gp->clear_input();
+  return true;
+}
+
+/// Performs constant folding test on the given model file \p NetFilename
+/// by replacing input tensors with name \p tensorNames, and values \p tensors
+/// and then checking against expected output expectedTensors. \returns true
+/// if the test completes without error.
+bool checkConstFoldedOutput(std::string NetFilename,
+                            llvm::ArrayRef<const char *> tensorNames,
+                            llvm::ArrayRef<Tensor *> tensors,
+                            llvm::ArrayRef<Tensor *> expectedTensors) {
+  ONNX_NAMESPACE::ModelProto modelDef;
+  std::string CFNetFilename = "dummy.onnxtxt";
+  auto lhsOrErrV = loadProto(NetFilename);
+  static_assert(IsLLVMExpected<decltype(lhsOrErrV)>(),
+                "Expected value to be a llvm::Expected");
+  if (!lhsOrErrV) {
+    return false;
+  }
+  modelDef = std::move(lhsOrErrV.get());
+  if (!replacePlaceholderWithConstant(modelDef, tensorNames, tensors))
+    return false;
+  if (!saveProto(CFNetFilename, modelDef))
+    return false;
+  setConstantFoldLoaderOpsFlag(true);
+  ExecutionEngine ee;
+  Module &mod = ee.getModule();
+  Function *F = mod.createFunction("temp");
+  ONNXModelLoader onnxLD(CFNetFilename, {}, {}, *F);
+  setConstantFoldLoaderOpsFlag(false);
+
+  const llvm::StringMap<Placeholder *> pOut = onnxLD.getOutputVarsMapping();
+  for (int i = 0; i < modelDef.graph().output_size(); i++) {
+    NodeValue nv;
+    auto lhsOrErrV =
+        onnxLD.getNodeValueByName(modelDef.graph().output(i).name());
+    static_assert(IsLLVMExpected<decltype(lhsOrErrV)>(),
+                  "Expected value to be a llvm::Expected");
+    if (!lhsOrErrV) {
+      return false;
+    }
+    nv = std::move(lhsOrErrV.get());
+    auto *constOut = llvm::dyn_cast<Constant>(nv.getNode());
+    EXPECT_TRUE(expectedTensors[i]->isEqual(constOut->getPayload()));
```

Comment:
You should `ASSERT_TRUE(constOut)` before this line to make sure it's actually a Constant.